Automattic/jetpack
# Assistant: Add timed discount card

#### Changes proposed in this Pull Request:

This PR introduces a new product recommendation card in the sidebar of the assistant used to help with Jetpack setup. It is not instantiated in this PR, hence this PR should not modify the current behaviour of the Assistant.

#### Jetpack product discussion


Context: pbNhbs-1xX-p2

#### Does this pull request change what data or activity we track or use?


Yes, this PR will fire a new `jetpack_recommendations_view_discounted_plans_button_click` event when the button on the `<DiscountCard />` component is clicked.

#### Testing instructions:






_Note: to reset the Assistant state, run the following commands:_
- `wp jetpack options delete recommendations_data`
- `wp jetpack options update recommendations_step not-started`

Regression testing:

- Run the plugin locally or spin up a JN site with this branch
- Make sure Jetpack is connected
- Visit `/wp-admin/admin.php?page=jetpack#/recommendations`
- Check that the Assistant looks and behaves as it does in the master branch

Discount card:

- Run the plugin locally
- Open the browser console
- Reset the Assistant state, following the aforementioned instructions
- In `projects/plugins/jetpack/_inc/client/recommendations/prompts/site-type/index.jsx`, make sure `canShowProductSuggestions` is `true` (line 128), and add a `discount` and `expiryDate` prop to the `DiscountCard` instance (line 113). E.g. `<DiscountCard discount={ 60 } expiryDate="2022-03-21 00:00:00"/>`
- Visit `/wp-admin/admin.php?page=jetpack#/recommendations`
- Check that the discount card shown in the right side of the Assistant looks like the mockup below (the CTA link doesn't matter for now)
- Make sure no exception was thrown in the console

<img width="441" alt="Screen Shot 2022-03-11 at 1 58 47 PM" src="https://user-images.githubusercontent.com/1620183/157935391-6dc4a5c7-334b-4c11-ab72-c89a2183c083.png">
